Distribution/Server/Util/CabalRevisions.hs

```diff
@@ -141,6 +140,32 @@ checkGenericPackageDescription
     withComponentName  f (name, condTree) = (name, (f name, condTree))
     withComponentName' f        condTree  = (f,             condTree)
 
+
+checkFlag :: Check Flag
+checkFlag flagOld flagNew = do
+    -- This check is applied via 'checkList' and for simplicity we
+    -- disallow renaming/reordering flags (even though reordering
+    -- would be fine semantically)
+    checkSame "Cannot change ordering of flags"
+              (flagName flagOld) (flagName flagNew)
+
+    -- Automatic flags' defaults may be changed as they don't make new
+    -- configurations reachable by the solver that weren't before
+    --
+    -- Moreover, automatic flags may be converted into manual flags
+    -- but not the other way round.
+    --
+    -- NB: We always allow to change the flag description as it has
+    --     purely informational value
```

Comment:
Need a `changesOk` for the description, otherwise these changes do not show up in the diff view.